reducers/storage.ts

```diff
@@ -110,6 +111,47 @@ function storage(state: Record<string, any> = {}, action: GenericAction) {
     }
 }
 
+function migrateDrafts(state: any) {
+    const drafts: any = {};
+    for (const storageKey in state) {
+        if (storageKey.startsWith('draft')) {
+            const storageDraft = state[storageKey];
+
+            // If draft does not have a channelId, it is a legacy draft and needs to be migrated
+            if (!storageDraft.value?.channelId) {
+                const info = getDraftInfoFromKey(storageKey, StoragePrefixes.DRAFT);
+                const timestamp = new Date(storageDraft.timestamp);
+
+                if (info === null || !info.id) {
+                    drafts[storageKey] = {timestamp, value: {message: '', fileInfos: [], uploadsInProgress: []}};
+                    continue;
+                }
+                const migratedDraft = {
+                    timestamp,
+                    value: {
+                        message: storageDraft.value?.message,
+                        fileInfos: storageDraft.value?.fileInfos || [],
+                        props: storageDraft.value?.props || {},
+                        uploadsInProgress: storageDraft.value?.uploadsInProgress || [],
+                        channelId: info.id,
+                        rootId: '',
+                        createAt: timestamp.getTime(),
+                        updateAt: timestamp.getTime(),
+                        show: true,
+                        remote: false,
+                    },
+                };
+
+                drafts[storageKey] = {...migratedDraft};
+            }
+        }
+    }
+
+    return {
+        ...drafts,
+    };
```

Comment:
More nitpicking, but I don't think we need to use destructuring to copy either of these since they're new objects we just made in this function. When we do that in other parts of these reducers, it's to copy things that are passed in (usually state) so that we aren't mutating something that was created outside of this function